R/weather_radars.R

```diff
@@ -0,0 +1,38 @@
+#' Get weather radar data
+#'
+#' The `weather_radars()` function is used to retreive weather radar metadata
+#' from the [OPERA network](https://www.eumetnet.eu/activities/observations-programme/current-activities/opera/).
+#' The retrieved data is returned as a tibble.
+#'
+#' @return A tibble containing weather radar metadata
+#' @export
+#'
+#' @examplesIf interactive()
+#' weather_radars()
+weather_radars <- function() {
+  # Build the url where the JSON file is hosted on Github
+  opera_radar_overview_url <-
+    paste(
+      sep = "/",
+      "https://raw.githubusercontent.com/enram",
+      "aloftdata.eu/refs/heads/main/_data",
+      "OPERA_RADARS_DB.json"
+    )
+
+  # Fetch the JSON file from Github with similar arguments as the other
+  # functions
+  httr2::request(opera_radar_overview_url) |>
+    req_user_agent_getrad() |>
+    httr2::req_retry(
+      max_tries = 15, backoff = \(x) sqrt(x) * 2,
+      is_transient = \(resp) httr2::resp_status(resp) %in% c(429),
+      retry_on_failure = TRUE
+    ) |>
+    httr2::req_perform() |>
+    # The object is actually returned as text/plain
+    httr2::resp_body_json(check_type = FALSE) |>
+    # As tibble so it displays more nicely
+    purrr::map(dplyr::as_tibble) |>
+    # Return as a single tibble by row binding
+    purrr::list_rbind()
```

Comment:
I would format some columns here as no one will uses longitude as a character. Otherwise I would always follow up by transforming myself. Maybe you could even make it an `sf`. Something like this:
```r
dplyr::mutate(latitude=as.numeric(latitude), longitude=as.numeric(longitude),band=ordered(band, levels=c('S','C','X')), country=factor(country) ) |>
```

Reply:
I agree on this fully, but it seems our source data can't even guarantee that latitude is a number. eg:

| number|country |countryid |oldcountryid | wmocode|odimcode |location        | status|latitude | longitude| heightofstation|band |doppler |polarization | maxrange|startyear | heightantenna|frequency | finishyear|source | diameterantenna| beam| gain|wrwp |singlerrr |compositerrr |wigosid      |
|------:|:-------|:---------|:------------|-------:|:--------|:---------------|------:|:--------|---------:|---------------:|:----|:-------|:------------|--------:|:---------|-------------:|:---------|----------:|:------|---------------:|----:|----:|:----|:---------|:------------|:------------|
|   1365|Iceland |BIRK43    |IL43         |      NA|isx1     |Huginn - mobile |      0|MOBILE   |        NA|              NA|X    |Y       |D            |      120|2012      |            NA|9.375     |         NA|main   |             1.9| 1.25| 42.4|NA   |NA        |NA           |0-352-6-isx1 |
|   1393|Iceland |BIRK44    |IL44         |      NA|isx2     |Muninn - mobile |      0|MOBILE   |        NA|              NA|X    |Y       |D            |      120|2013      |            NA|9.355     |         NA|main   |             1.9| 1.25| 42.4|NA   |NA        |NA           |0-352-6-isx2 |

Reply:
Types for the whole table: 

![image](https://github.com/user-attachments/assets/39889184-4829-44dd-805b-668b32703d82)

Reply:
I guess this goes back to discussions we had in Braunschweig and having some influence the source data. Which were were discussing with @peterdesmet and @leijnse . One option would be to have an output format option so you can ask for a more curated version or raw. I'm also not against formatting it already in the hope this gets fixed in the source.

Reply:
The source file could definitely be improved: all values are currently strings for example. I think we should cast those to more reasonable types (especially numbers). Values that don't fit should silently return NA.

Regarding raw vs curated: this makes me realize that it would actually be better if the function:

- Used the actual (2!) source files from http://eumetnet.eu/wp-content
- [Combined those into one](https://github.com/enram/aloftdata.eu/blob/7efdbbfb18e7b3e811d502d9b8e7b7f51421c0f9/_data/OPERA_RADARS_DB.R#L21)
- Does some casting to NA and numbers
- Returns a dataframe

I would then use that function to create https://github.com/enram/aloftdata.eu/blob/main/_data/OPERA_RADARS_DB.json (and remove the script at https://github.com/enram/aloftdata.eu/blob/main/_data/OPERA_RADARS_DB.R). I typically inspect the git changes and report any errors

I am not sure a `raw` version of the function is necessary.

Reply:
I did actually consider this, but the Issue specifically referred to this file :sweat_smile: , using the actual source and casting seems like a good idea. 

Should values that don't fit not offer a warning and then be returned as NA? 

In the case above, it might be good to know the radar was mobile.

Reply:
> Should values that don't fit not offer a warning and then be returned as NA?

The user won't be able to do anything about those warnings though, so a bit annoying to get them.

> In the case above, it might be good to know the radar was mobile.

I think we should move this information to a `comments` field as part of our processing.